rust/arrow/src/compute/util.rs

```diff
@@ -44,10 +45,53 @@ where
     }
 }
 
+/// Takes/filters a list array's inner data using the offsets of the list array.
+///
+/// Where a list array has index `[0,2,5,10]`, taking an index of `[2,0]` returns
+/// an array of the indices `[5..10, 0..2]` and offsets `[0,5,7]` (5 elements and 2
+/// elements)
+pub(crate) fn take_index_from_list(
```

Comment:
- `pub(crate` could be changed to `pub(super)` if this is only used within the `compute` module? or, is it better to put this in `take.rs`?
- the name is a little bit misleading - it gives the impression that this is taking indices from a list - maybe rename to `take_value_indices_from_list` or something else?

Reply:
Can we keep it here for now? I wanted to use it in a `sort` kernel, which would sort an array and return indices; then use this function (for lists) to take the appropriate value indices.